Coding/fw/lib/Settings/WLAN.cpp

```diff
@@ -0,0 +1,211 @@
+}
+
+/**************************************************************************************************/
+const String &wlan::getAP_SSID()
+{
+    return AP_SSID;
+}
+
+/**************************************************************************************************/
+const String &wlan::getAP_PASS()
+{
+    return AP_PASSWORD;
+}
+
+/**************************************************************************************************/
+const bool &wlan::getAP_MODE()
+{
+    return m_APMode;
+}
+
+/**************************************************************************************************/
+bool wlan::begin()
+{
+    bool success = true;
+    Board::apLED.write(HIGH);
+    Board::staLED.write(HIGH);
+
+    Settings::get(DIRECTORY, "STA_SSID", STA_SSID, "");
+    Settings::get(DIRECTORY, "STA_Password", STA_PASSWORD, "");
+
+    readWiFiMode();
+    if (m_APMode)
+    {
+        Board::staLED.write(LOW);
+        if (!WiFi.softAP(AP_SSID.c_str(), AP_PASSWORD.c_str()))
+        {
+            success = false;
+        }
+
+        m_serverIP = WiFi.softAPIP();
+    }
+    else
+    {
+        Board::apLED.write(LOW);
+        if (!WiFi.mode(WIFI_STA))
+        {
+            success = false;
+        }
+        else if (WiFi.begin(STA_SSID.c_str(), STA_PASSWORD.c_str()) == WL_CONNECT_FAILED)
+        {
+            success = false;
+        }
+        else if (!connectWiFi())
+        {
+            success = false;
+        }
+
+        m_serverIP = WiFi.localIP();
+    }
+    return success;
+}
+
+/**************************************************************************************************/
+/**
+*   Calls next request on DNS Server
+*   return true if a restart is requested
+*/
+bool wlan::checkConnection()
+{
+    bool success = true;
+    if (WiFi.getMode() == WIFI_STA && WiFi.status() != WL_CONNECTED)
+    {
+        if (!connectWiFi())
+        {
+            success = false;
+        }
+    }
+    return success;
+}
+
+IPAddress wlan::getIPAddress()
+{
+    return m_serverIP;
+}
+
+void wlan::saveConfig(const String &key, const String &value)
+{
+    Settings::save(DIRECTORY, key, value);
+}
+
+/* PROTECTED METHODES *****************************************************************************/
+
+/* PRIVATE METHODES *******************************************************************************/
+
+/* EXTERNAL FUNCTIONS *****************************************************************************/
+
+/* INTERNAL FUNCTIONS *****************************************************************************/
+/**************************************************************************************************/
+static void readWiFiMode()
+{
+    uint8_t currentBtnState = LOW;
+    uint8_t previousBtnState = LOW;
+    uint32_t lastDebounceTime = 0;
+
+    const uint8_t DEBOUNCE_DELAY = 50;
+    const uint32_t SETUP_TIME = 2000;
+    const uint32_t START_TIME = millis();
+
+    while ((millis() - START_TIME) < SETUP_TIME)
```

Comment:
do you really need to wait the full 2 seconds, or until a stable input was seen for 50ms ? 
What happens if there is no stable input  in the 2 second intervall ? THere should be an error halt in my opinion.